apps/climatemappedafrica/src/components/DataIndicators/DataIndicators.js

```diff
@@ -0,0 +1,133 @@
+import {
+  ClickAwayListener,
+  Grid,
+  Dialog,
+  Slide,
+  useMediaQuery,
+} from "@mui/material";
+import { useTheme } from "@mui/material/styles";
+import clsx from "clsx";
+import PropTypes from "prop-types";
+import React, { useState } from "react";
+
+import Icon from "./Icon";
+import IndicatorPanel from "./IndicatorPanel";
+import useStyles from "./useStyles";
+
+import bg from "@/climatemappedafrica/assets/images/Mask Group 8.png";
+import Header from "@/climatemappedafrica/components/Header";
+import Image from "@/climatemappedafrica/components/Image";
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="left" timeout={300} ref={ref} {...props} />;
+});
+
+function DataIndicators({ indicators, title, ...props }) {
+  const classes = useStyles(props);
+  const theme = useTheme();
+  const isDesktop = useMediaQuery(theme.breakpoints.up("lg"));
+
+  const [checked, setChecked] = useState(false);
+  const [currentItemIndex, setCurrentItemIndex] = useState(null);
+
+  if (!indicators?.length) {
+    return null;
+  }
+  const handleIconClick = (index) => {
+    setCurrentItemIndex(index);
+    setChecked(true);
+  };
+  const resetItemClick = () => {
+    setChecked(false);
+    setCurrentItemIndex(null);
+  };
+  const currentItem = indicators[currentItemIndex];
+  const panelProps = isDesktop
+    ? {
+        in: checked,
+        mountOnEnter: true,
+        unmountOnExit: true,
+        component: Slide,
+        direction: "left",
+        timeout: 300,
+        classes: {
+          root: classes.slide,
+          content: classes.content,
+          title: classes.title,
+          description: classes.description,
+        },
+      }
+    : {
+        open: checked,
+        onClose: resetItemClick,
+        component: Dialog,
+        BackdropProps: {
+          classes: {
+            root: classes.backdrop,
+          },
+        },
+        TransitionComponent: Transition,
+        classes: {
+          root: classes.dialog,
+          paper: classes.dialogPaper,
+          content: classes.content,
+          title: classes.title,
+          description: classes.description,
+        },
+      };
+
+  return (
+    <div className={classes.root}>
+      <div className={classes.background}>
+        <Image objectFit="cover" src={bg} layout="fill" />
+      </div>
+      <div className={classes.section}>
+        <div
+          className={clsx(classes.indicatorsContainer, {
+            [classes.slideIn]: checked,
+          })}
+        >
+          <Header className={classes.header}>{title}</Header>
+          <ClickAwayListener onClickAway={resetItemClick}>
+            <Grid container alignItems="center" justifyContent="center">
+              {indicators?.map((item, index) => (
```

Comment:
```suggestion
              {indicators.map((item, index) => (
```
You've already checked for `indications?.length` up top.